tsdb/block_test.go

```diff
@@ -303,7 +304,13 @@ func TestReadIndexFormatV1(t *testing.T) {
 
 // createBlock creates a block with given set of series and returns its dir.
 func createBlock(tb testing.TB, dir string, series []Series) string {
-	return createBlockFromHead(tb, dir, createHead(tb, series))
+	w, close := newTestNoopWal(tb)
+	defer close()
+	head := createHead(tb, series, w)
+	defer func() {
+		testutil.Ok(tb, head.Close())
+	}()
+	return createBlockFromHead(tb, dir, head)
```

Comment:
This refactoring is to close the Head only after the block is created. Closing first and then accessing the data will cause problem for mmapped chunks.

Reply:
This is quite complex. Why `createHead` cannot close it properly?

Reply:
If `createHead` closes the Head, then the mmap chunk reader is also closed and we can no longer read the mmapped chunks again. Hence we need to close the Head only after finishing all the operations with the Head.